### PR TITLE
[4.0] HTML user selection

### PR DESCRIPTION
**Pull request by priiish**
Pull Request for Issue #22497  .

### Summary of Changes
Changed link into button tag so there won't be any difficulties with spacebar usage.
Also no div is created within a span element anymore (invalid html).

### Testing Instructions
Have a look at the article menu ("publishing"). Next to "Created By" is the button to select a user.

### Expected result
Button is now within a button tag, the div no longer within the span element.

### Actual result
As expected.

### Documentation Changes Required
No.

### CODE REVIEW

File: layouts/joomla/form/field/user.php [hunk 1]
```diff
@@ -111,22 +111,22 @@
...
-						'title'       => Text::_('JLIB_FORM_CHANGE_USER'),
-						'closeButton' => true,
-						'height'      => '100%',
-						'width'       => '100%',
-						'modalWidth'  => 80,
-						'bodyHeight'  => 60,
-						'footer'      => '<a type="button" class="btn btn-secondary" data-dismiss="modal">' . Text::_('JCANCEL') . '</a>'
-					)
-				); ?>
+					<button class="btn btn-primary button-select" title="<?php echo Text::_('JLIB_FORM_CHANGE_USER') ?>"><span class="fa fa-user icon-white" aria-hidden="true"></span></button>
```
Comment (brianteeman) [inline layouts/joomla/form/field/user.php]: need to add type="button"

Comment (priiish) [reply layouts/joomla/form/field/user.php]: fixed that

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5775862467d298a8d0c7cf17b40a62cc58efa011<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22878">issues.joomla.org/tracker/joomla-cms/22878</a>.</sub>

File: layouts/joomla/form/field/user.php [hunk 2]
```diff
@@ -111,22 +111,22 @@
...
-						'width'       => '100%',
-						'modalWidth'  => 80,
-						'bodyHeight'  => 60,
-						'footer'      => '<a type="button" class="btn btn-secondary" data-dismiss="modal">' . Text::_('JCANCEL') . '</a>'
-					)
-				); ?>
+					<button type="button" class="btn btn-primary button-select" title="<?php echo Text::_('JLIB_FORM_CHANGE_USER') ?>"><span class="fa fa-user icon-white" aria-hidden="true"></span></button>
 				</span>
+            <?php echo HTMLHelper::_(
+                'bootstrap.renderModal',
```
Comment (laoneo) [inline layouts/joomla/form/field/user.php:117]: Can you convert spaces to tabs please?

**Comment by PhilETaylor** [comment]
after appling this patch the html rendered is:

```html
<div class="input-group">
		<input type="text" id="jform_created_by" class="form-control field-user-input-name" value="" placeholder="Select a User" readonly="">
					<span class="input-group-append">
					<button type="button" class="btn btn-primary button-select" title="Select User"><span class="fa fa-user icon-white" aria-hidden="true"></span></button>
				</span>
            <div id="userModal_jform_created_by" role="dialog" tabindex="-1" class="joomla-modal modal fade" data-url="index.php?option=com_users&amp;view=users&amp;layout=modal&amp;tmpl=component&amp;required=0&amp;field=jform_created_by" data-iframe="<iframe class=&quot;iframe&quot; src=&quot;index.php?option=com_users&amp;view=users&amp;layout=modal&amp;tmpl=component&amp;required=0&amp;field=jform_created_by&quot; name=&quot;Select User&quot; height=&quot;100%&quot; width=&quot;100%&quot;></iframe>">
	<div class="modal-dialog modal-lg jviewport-width80" role="document">
		<div class="modal-content">
			<div class="modal-header">
			<h3 class="modal-title">Select User</h3>
				<button type="button" class="close novalidate" data-dismiss="modal" aria-label="Cancel">
			<span aria-hidden="true">×</span>
		</button>
	</div>
<div class="modal-body jviewport-height60">
	</div>
<div class="modal-footer">
	<a type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</a></div>
		</div>
	</div>
</div>
			</div>
```

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on 5775862467d298a8d0c7cf17b40a62cc58efa011<br /><br />applied patch - it does as it says, no errors, could activate with tab/space<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22878">issues.joomla.org/tracker/joomla-cms/22878</a>.</sub>

**Comment by Quy** [comment]
Replaced by PR #25206<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22878">issues.joomla.org/tracker/joomla-cms/22878</a>.</sub>

**Pull request closed without merging**